src/norm_flows.jl

```diff
@@ -0,0 +1,139 @@
+using Distributions
+using LinearAlgebra
+using Random
+using StatsFuns: softplus
+using Roots, LinearAlgebra # for inverse
+
+################################################################################
+#                            Planar and Radial Flows                           #
+#             Ref: Variational Inference with Normalizing Flows,               #
+#               D. Rezende, S. Mohamed(2015) arXiv:1505.05770                  #
+################################################################################
+
+mutable struct PlanarLayer{T1,T2} <: Bijector
+    w::T1
+    u::T1
+    u_hat::T1
```

Comment:
I think we don't need to store `u_hat` in the filed at all.
We just create `u_hat` whennever it is needed during computation.